mikrotik/resource_bgp_instance_test.go

```diff
@@ -0,0 +1,226 @@
+		Steps: []resource.TestStep{
+			{
+				Config: testAccBgpInstance(),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccBgpInstanceExists(resourceName),
+					resource.TestCheckResourceAttrSet(resourceName, "id")),
+			},
+			{
+				PreConfig:          removeBgpInstance,
+				Config:             testAccBgpInstance(),
+				ExpectNonEmptyPlan: false,
+			},
+		},
+	})
+}
+
+func TestAccMikrotikBgpInstance_updateBgpInstance(t *testing.T) {
+	resourceName := "mikrotik_bgp_instance.bar"
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckMikrotikBgpInstanceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccBgpInstance(),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccBgpInstanceExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "name", originalBgpName),
+					resource.TestCheckResourceAttr(resourceName, "as", originalAs),
+					resource.TestCheckResourceAttr(resourceName, "router_id", originalRouterId),
+				),
+			},
+			{
+				Config: testAccBgpInstanceUpdatedAsAndRouterId(),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccBgpInstanceExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "name", originalBgpName),
+					resource.TestCheckResourceAttr(resourceName, "as", updatedAs),
+					resource.TestCheckResourceAttr(resourceName, "router_id", updatedRouterId),
+				),
+			},
+			{
+				Config: testAccBgpInstanceUpdatedOptionalFields(),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccBgpInstanceExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "name", originalBgpName),
+					resource.TestCheckResourceAttr(resourceName, "as", updatedAs),
+					resource.TestCheckResourceAttr(resourceName, "router_id", updatedRouterId),
+					resource.TestCheckResourceAttr(resourceName, "comment", commentBgpInstance),
+					resource.TestCheckResourceAttr(resourceName, "cluster_id", originalClusterId),
+					resource.TestCheckResourceAttr(resourceName, "client_to_client_reflection", "false"),
+					resource.TestCheckResourceAttr(resourceName, "confederation", originalConfederation),
+				),
+			},
+		},
+	})
+}
+
+func TestAccMikrotikBgpInstance_import(t *testing.T) {
+	resourceName := "mikrotik_bgp_instance.bar"
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckMikrotikBgpInstanceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccBgpInstance(),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccBgpInstanceExists(resourceName),
+					resource.TestCheckResourceAttrSet(resourceName, "id")),
+			},
+			// TODO:  figure out why this fails
+			{
+				ResourceName: resourceName,
+				// tried adding this field, but didn't help
+				ImportStateId:     originalBgpName,
```

Comment:
You shouldn't need this except for more custom cases. I'm not sure if it applies or not in this situation but I personally have never needed it.

Can you please paste the failed test output with `TF_LOG=debug` set?

Reply:
Ah, sorry this was leftover from when I was initially having trouble with my `import` acceptance test (the issue we talked about offline which was due to the buggy way I had set the object id in tf state).